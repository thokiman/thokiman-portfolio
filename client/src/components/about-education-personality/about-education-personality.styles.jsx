import styled from "styled-components";

export const AboutPersonalityContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: 1fr;
  width: 100%;
  margin: 0 auto;
  @media only screen and (max-width: 800px) {
    grid-template-columns: repeat(3, 1fr);
  }
`;

export const AboutPersonalityTitle = styled.div`
  text-align: center;
  padding: 40px 10px;
  font-size: 250%;
  color: #e3e1e4;
  @media only screen and (max-width: 800px) {
    font-size: 7vw;
  }
`;

export const AboutPersonalityBox = styled.div`
  opacity: 0.6;
  width: 100%;
  transition: background-color 0.6s ease-in-out;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const AboutPersonalityTextBox = styled.div`
  padding: 0 5px;
  opacity: 1;
  text-align: center;
  font-size: 120%;
  transition: color 0.2s ease-in-out;
  font-kerning: auto;
  letter-spacing: 0.025em;
  @media only screen and (max-width: 800px) {
    font-size: 5vw;
  }
`;

AboutPersonalityContainer.displayName = "AboutPersonalityContainer";
AboutPersonalityTitle.displayName = "AboutPersonalityTitle";
AboutPersonalityBox.displayName = "AboutPersonalityBox";
AboutPersonalityTextBox.displayName = "AboutPersonalityTextBox";
