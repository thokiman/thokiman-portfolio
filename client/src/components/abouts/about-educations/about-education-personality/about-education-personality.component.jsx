import React, { useEffect, useState } from 'react';
import { getRandomColor } from '../../../../utils/about.component.utils';
import {
  AboutPersonalityBox,
  AboutPersonalityContainer,
  AboutPersonalityTextBox,
  AboutPersonalityTitle,
} from './about-education-personality.styles';

export const AboutEducationPersonality = ({ personality }) => {
  const [backgroundColor, setBackgroundColor] = useState({
    1: '',
    2: '',
    3: '',
    4: '',
    5: '',
    6: '',
    7: '',
    8: '',
    9: '',
    10: '',
    11: '',
    12: '',
    13: '',
    14: '',
    15: '',
    16: '',
  });
  const [color, setColor] = useState({
    1: '',
    2: '',
    3: '',
    4: '',
    5: '',
    6: '',
    7: '',
    8: '',
    9: '',
    10: '',
    11: '',
    12: '',
    13: '',
    14: '',
    15: '',
    16: '',
  });

  useEffect(() => {
    const timerId = setInterval(() => {
      getRandomBackgroundColor();
    }, 400);

    return () => clearInterval(timerId);
  });

  const getRandomBackgroundColor = () => {
    const h_range = [0, 360];
    const s_range = [30, 40];
    const l_range = [25, 25];
    const a_range = [1, 1];
    const color = getRandomColor(h_range, s_range, l_range, a_range);

    const h_range_background_color = [0, 360];
    const s_range_background_color = [80, 90];
    const l_range_background_color = [60, 75];
    const a_range_background_color = [1, 1];

    const backgroundColor = getRandomColor(
      h_range_background_color,
      s_range_background_color,
      l_range_background_color,
      a_range_background_color
    );
    setBackgroundColor({
      1: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      2: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      3: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      4: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      5: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      6: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      7: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      8: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      9: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      10: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      11: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      12: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      13: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      14: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      15: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });
    setBackgroundColor({
      16: `hsla(${backgroundColor.h + Math.random() * backgroundColor.h}, ${
        backgroundColor.s
      }%, ${backgroundColor.l + Math.random() * backgroundColor.l}%, ${
        backgroundColor.a
      })`,
    });

    setColor({
      1: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      2: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      3: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      4: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      5: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      6: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      7: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      8: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      9: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      10: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      11: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      12: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      13: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      14: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      15: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
    setColor({
      16: `hsla(${color.h + Math.random() * color.h}, ${color.s}%, ${
        color.l + Math.random() * color.l
      }%, ${color.a})`,
    });
  };

  return (
    <AboutPersonalityContainer className='about-personality-content'>
      <AboutPersonalityTitle>Personality</AboutPersonalityTitle>
      {personality.map(({ id }, i) => {
        return (
          <AboutPersonalityBox
            className='about-personality-box'
            key={id}
            style={{
              color: `${color[i + 1]}`,
              backgroundColor: `${backgroundColor[i + 1]}`,
            }}
          >
            <AboutPersonalityTextBox>
              {personality[i].description}
            </AboutPersonalityTextBox>
          </AboutPersonalityBox>
        );
      })}
    </AboutPersonalityContainer>
  );
};

export default AboutEducationPersonality;
