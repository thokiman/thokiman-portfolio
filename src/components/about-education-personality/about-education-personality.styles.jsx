import styled from "styled-components";

export const AboutPersonalityContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-auto-rows: 1fr;
  width: 100%;
  margin: 0 auto;
`;

export const AboutPersonalityTitle = styled.div`
  text-align: center;
  padding: 40px 10px;
  font-size: 250%;
  color: #e3e1e4;
`;

export const AboutPersonalityBox = styled.div`
  opacity: 0.6;
  width: 100%;
  transition: background-color 0.6s ease-in-out;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const AboutPersonalityTextBox = styled.div`
  padding: 0 5px;
  opacity: 1;
  text-align: center;
  font-size: 120%;
  transition: color 0.2s ease-in-out;
  font-kerning: auto;
  letter-spacing: 0.025em;
`;

AboutPersonalityContainer.displayName = "AboutPersonalityContainer";
AboutPersonalityTitle.displayName = "AboutPersonalityTitle";
AboutPersonalityBox.displayName = "AboutPersonalityBox";
AboutPersonalityTextBox.displayName = "AboutPersonalityTextBox";
