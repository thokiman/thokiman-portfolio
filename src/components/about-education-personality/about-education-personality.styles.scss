.personality-container {

    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-auto-rows: 1fr;

    width: 100%;
    margin: 0 5px;

    .personality-title {
        text-align: center;
        padding: 40px 10px;
        font-size: 250%;

        color: #e3e1e4
    }

    .personality-box {
        opacity: 0.6;
        width: 100%;
        transition: background-color 0.6s ease-in-out;



        .text-box {
            opacity: 1;
            text-align: center;
            padding: 40px 7px;
            font-size: 190%;
            transition: color 0.2s ease-in-out;
            font-kerning: auto;
            letter-spacing: 0.025em;
        }
    }
}