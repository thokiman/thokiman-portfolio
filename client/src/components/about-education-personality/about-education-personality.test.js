import React from "react";
import { shallow } from "enzyme";

import { AboutEducationPersonality } from "./about-education-personality.component";

describe("AboutEducationPersonality component", () => {
  let wrapper;
  const mockArray = [];

  beforeEach(() => {
    const mockProps = {
      personality: mockArray,
    };
    wrapper = shallow(<AboutEducationPersonality {...mockProps} />);
  });

  it("should render AboutEducationPersonality component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
