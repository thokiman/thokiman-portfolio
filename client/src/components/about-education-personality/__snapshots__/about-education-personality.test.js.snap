// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutEducationPersonality component should render AboutEducationPersonality component 1`] = `ShallowWrapper {}`;
